modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AMQPConstructStore.java

```diff
@@ -122,15 +122,26 @@ public void addQueue(AndesQueue queue, boolean isLocal) throws AndesException {
      * remove a queue
      *
      * @param queueName name of the queue to be removed
-     * @param isLocal   is this a local change
      * @throws AndesException
      */
-    public void removeQueue(String queueName, boolean isLocal) throws AndesException {
-        if (isLocal) {
-            andesContextStore.deleteQueueInformation(queueName);
-            //create the space created to keep message counter on this queue
-            messageStore.removeQueue(queueName);
-        }
+    public void removeLocalQueue(String queueName) throws AndesException {
+
+        // Remove the queue from internal maps
+        removeClusterQueue(queueName);
```

Comment:
I'm sorry I don't get how invalidating the cache afterwards could avoid race conditions.

The rationale behind removing it beforehand was that the queue removal should be reflected in the memory before it gets removed from the store, or else if somebody tries to publish a message to a queue which is deleted from the db but not yet from the memory, there will be problems.